### PR TITLE
feat(backend): forbid usage of group mentions in usernames

### PR DESCRIPTION
**Pull request by grignards**
See #3328

## Checkpoints

These points must be checked before merging. Please don't edit them out.

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

**Testing method**

Try to create a user with `all`/`tous`/`todos` as username: this should fail with a 400 error. Same with modifying an existing username to those values.

### CODE REVIEW

File: backend/tracim_backend/error.py [hunk 1]
```diff
@@ -76,6 +76,7 @@ class ErrorCode(IntEnum):
     USERNAME_ALREADY_EXIST_IN_DB = 2063
```
Comment (raphj) [inline backend/tracim_backend/error.py:76]: Maybe rename this to `USERNAME_ALREADY_EXISTS_IN_DB` (or even `USERNAME_ALREADY_EXISTS`)?

Comment (grignards) [reply backend/tracim_backend/error.py]: Done

**Comment by inkhey** [comment]
@grignards What is the behaviour is an user with all as username already exist ?

**Comment by grignards** [comment]
> @grignards What is the behaviour is an user with all as username already exist ?

The user won't get any mention as the group mentions take precedence. The user can always change his username to a non-reserved one.

**Comment by gbonaspetti** [comment]
Tested on branch

**Pull request merged**